core/src/commonTest/kotlin/operations/numeric/SubtractionTest.kt

```diff
@@ -43,7 +43,7 @@ class SubtractionTest : FunSpec({
             ),
             TestInput(
                 expression = mapOf("-" to listOf("1", 1, listOf("banana"))),
-                result = JsonLogicResult.Success(0.0)
+                result = JsonLogicResult.Success(0)
```

Comment:
The tests use non-strict comparisons - [source](https://github.com/kotest/kotest/blob/master/kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/NumberEq.kt) 

We could use strict comparisons to avoid confusion / mistakes, but then we would have to make a copy-paste of a kotest function `shouldBe`, and add `strictNumberEq = true` to `eq(actual, expected)?.let(errorCollector::collectOrThrow)`

This is too close to inner workings of the library for my liking :/ afaik there is no `shouldBeStrict` or something like that to easily use here. + If we normalize to Long type then we would need to change all of the test data from 0 to 0L for example :')

Reply:
In this case clients will be responsible for checking if the returned value is OK for them. Contract is based on Any return type so this change won't break anything.